### ISSUE TITLE
Detecting small objects in large images with DetectNet

### ISSUE THREAD

**Issue by DarylWM** [NVIDIA/DIGITS#1062]
Hi.

My images are false colour images of mass spectrometer data. Each one is 3292 x 415456 pixels. The objects I want to detect vary in size from 7 x 5 to 18 x 11. I'm using 4 x K520 GPUs (i.e. 4GB GPU memory each).

From reading the guidance [here](https://github.com/NVIDIA/DIGITS/issues/980), I'm trying these steps:
* Take subset images of 1000 x 1000 (to make them fit in the GPU's memory).
* Change all the image sizes in detectnet_network.prototxt (except train_transform) to 1000 x 1000.
* Change image_size in train_transform to 500 x 500 and crop_bboxes to false.
* Scale my training images up so the objects are at least 50 x 50 pixels.

Do I also need to scale up my test images, so the model can detect the objects?
To avoid scaling up the images, would I better to change the stride?

Thanks.

**Comment by lukeyeager** [comment]
I'm not following how you're changing the sizes of your images. Here are the ways the image size can be changed:

* `Data` layer
  * You can specify a `crop_size` to crop the data down to `CROP_SIZE x CROP_SIZE`
    * Don't do this - the labels won't be adjusted
* `DetectNetTransformation` layer
  * `TRAIN` phase
    * If `rand() > scale_prob`, resize the images to something in [`scale_min`, `scale_max`]
    * Crop the resized result to [`image_size_x`,`image_size_y`]
  * `TEST` phase
    * Images are always resized to [`image_size_x`,`image_size_y`]

With that explanation, can you tell me how you're doing this?

> Scale my training images up so the objects are at least 50 x 50 pixels.

**Comment by DarylWM** [comment]
Hi Luke. On re-reading my question I can see it's unclear how I'm trying to achieve this.

To satisfy the constraints of (i) fitting a training set image for DetectNet in GPU memory (I've found the max dimensions to be 1000x1000 on a K520), and (ii) having objects larger than 50x50px, in **pre-processing** I am proposing these steps:

1. From my original 3292x415456 image, for each object in the training list of object bounding boxes, crop a 100x100 area with top-left coordinates the same as the object's bounding box. Resize this image by 10 (so the objects become at least 50x50) to get a 1000x1000 image, and save that in 000n.png.
2. From the bounding boxes, work out if there are other objects in that cropped 100x100 area.
3. Adjust the bounding box coordinates and save them in a 000n.txt file.

This would form my training set for the DetectNet model, which I have modified to accept 1000x1000 images, by:

1. Changing all the image sizes in detectnet_network.prototxt (except train_transform) to 1000 x 1000.
2. Changing image_size in train_transform to 500 x 500 and crop_bboxes to false.

Hopefully that's clearer. My question is whether this is the most sensible approach.
